plugins/hypemachine/integration.js

```diff
@@ -0,0 +1,133 @@
+function update() {
+    return {
+        "playbackStatus": getPlaybackStatus(),
+        "canSeek": false,
+        "canGoNext": hasNext(),
+        "canGoPrevious": hasPrevious(),
+        "canAddToFavorites": true,
+        "volume": 1,
+        "duration": readTime("player-time-total"),
+        "position": readTime("player-time-position"),
+        "songId": getSongId(),
+        "songTitle": getSongTitle(),    
+        "artistName": getArtistName(),
+        "albumTitle": getAlbumTitle(),
+        "artUrl": getAlbumArt(),
+        "isFavorite": isFavorite()
+    };
+}
+
+function isFavorite() {
+    return hasClass(document.getElementById("playerFav"), "fav-on");
+}
+
+function hasNext() {
```

Comment:
This function is useless, better set the property to true directly. Unless you planned to implement detection of next/previous songs?